climada/hazard/centroids/centr.py

```diff
@@ -1244,152 +813,103 @@ def _from_hdf5(cls, data):
                     meta[key] = value[0]
                 else:
                     meta[key] = rasterio.Affine(*value)
-            centr = cls(meta=meta)
+            latitude, longitude = _meta_to_lat_lon(meta)
 
+        extra_values = {}
         for centr_name in data.keys():
-            if centr_name not in ('crs', 'lat', 'lon', 'meta'):
-                setattr(centr, centr_name, np.array(data.get(centr_name)))
-        return centr
-
-    @property
-    def crs(self):
-        """Get CRS of raster or vector."""
-        if self.meta:
-            return self.meta['crs']
-        if self.geometry.crs:
-            return self.geometry.crs
-        return DEF_CRS
+            if centr_name not in ('crs', 'lat', 'lon', 'meta', 'latitude', 'longitude'):
+                values = np.array(data.get(centr_name))
+                if latitude.size != 0 and values.size != 0 :
+                    extra_values[centr_name] = values
+
+        return gpd.GeoDataFrame(
+            extra_values,
+            geometry=gpd.points_from_xy(x=longitude, y=latitude, crs=crs)
+        )
 
-    @property
-    def size(self):
-        """Get number of pixels or points."""
-        if self.meta:
-            return int(self.meta['height'] * self.meta['width'])
-        return self.lat.size
+    '''
+    Private methods
+    '''
 
-    @property
-    def shape(self):
-        """Get shape of rastered data."""
-        try:
-            if self.meta:
-                return (self.meta['height'], self.meta['width'])
-            return (np.unique(self.lat).size, np.unique(self.lon).size)
-        except AttributeError:
-            return ()
 
-    @property
-    def total_bounds(self):
-        """Get total bounds (left, bottom, right, top)."""
-        if self.meta:
-            left = self.meta['transform'].xoff
-            right = left + self.meta['transform'][0] * self.meta['width']
-            if left > right:
-                left, right = right, left
-            top = self.meta['transform'].yoff
-            bottom = top + self.meta['transform'][4] * self.meta['height']
-            if bottom > top:
-                bottom, top = top, bottom
-            return left, bottom, right, top
-        return self.lon.min(), self.lat.min(), self.lon.max(), self.lat.max()
+    def _ne_crs_geom(self):
+        """Return `geometry` attribute in the CRS of Natural Earth.
 
-    @property
-    def coord(self):
-        """Get [lat, lon] array."""
-        return np.stack([self.lat, self.lon], axis=1)
+        Returns
+        -------
+        geo : gpd.GeoSeries
+        """
+        if u_coord.equal_crs(self.gdf.crs, u_coord.NE_CRS):
+            return self.gdf.geometry
+        return self.gdf.geometry.to_crs(u_coord.NE_CRS)
 
-    def set_geometry_points(self, scheduler=None):
-        """Set `geometry` attribute with Points from `lat`/`lon` attributes.
+    def _set_region_id(self, level='country', overwrite=False):
```

Comment:
Why is this method private? Shouldn't users should be able to set the region ID after initialization?

Reply:
No, we decided that data should NOT be modified in place after initialization. It is done in the init if desired by the user.

Reply:
Alright, then new proposal: Make this method a free function, which takes a geometry series as (additional) argument and returns a series of corresponding region IDs. The constructor of Centroids then may call this function (depending on the init arguments), and users are free to do so, too. Same for `_set_on_land`.

I see no reason why this function should be private to Centroids.

Reply:
This is already what the method does `u_coord.get_country_code` does (not on geometries, but on lat/lon coordinates). The `_set_region_id` is here just a convenience method to make the calls in the `_init_` simpler. It is private because it is intended to be called only privately.